src/server/frontend/wayland/wayland_connector.cpp

```diff
@@ -1686,8 +1687,14 @@ class WlAbstractMirWindow : public WlMirWindow
     mf::SurfaceId surface_id;
 
 private:
-    void commit(geom::Size const& buffer_size) override
+    void commit(geom::Size const& buffer_size_) override
     {
+        auto buffer_size = buffer_size_;
+
+        // Sometimes, when using xdg-shell, qterminal creates an insanely tall buffer
+        if (buffer_size.height > geom::Height{10000})
+            buffer_size.height = geom::Height{1000};
```

Comment:
Also, I happen to know that sometimes Qt creates windows with junk dimensions, but sets them only before making them visible. That might be what's happening?

Reply:
I dislike this too I just did this to prevent memory being exhausted, I've no objection to logging a warning here. (But I still hope for a better solution before enabling by default - maybe some spelunking through the Qt code will be inspiring.)

Note that Qt has submitted a buffer here, this is not simply failing to set dimensions before use.

Reply:
Hmm, peculiar behaviour by Qt then